pandas/tests/io/test_orc.py

```diff
@@ -224,3 +248,61 @@ def test_orc_reader_snappy_compressed(dirpath):
     got = read_orc(inputfile).iloc[:10]
 
     tm.assert_equal(expected, got)
+
+
+@td.skip_if_no("pyarrow", min_version="7.0.0")
+def test_orc_roundtrip_file(dirpath):
+    # GH44554
+    # PyArrow gained ORC write support with the current argument order
+    data = {
+        "boolean1": np.array([False, True], dtype="bool"),
+        "byte1": np.array([1, 100], dtype="int8"),
+        "short1": np.array([1024, 2048], dtype="int16"),
+        "int1": np.array([65536, 65536], dtype="int32"),
+        "long1": np.array([9223372036854775807, 9223372036854775807], dtype="int64"),
+        "float1": np.array([1.0, 2.0], dtype="float32"),
+        "double1": np.array([-15.0, -5.0], dtype="float64"),
+        "bytes1": np.array([b"\x00\x01\x02\x03\x04", b""], dtype="object"),
+        "string1": np.array(["hi", "bye"], dtype="object"),
+    }
+    expected = pd.DataFrame.from_dict(data)
+
+    outputfile = os.path.join(dirpath, "TestOrcFile.testReadWrite.orc")
+    expected.to_orc(outputfile)
+    got = read_orc(outputfile)
+
+    tm.assert_equal(expected, got)
+
+
+@td.skip_if_no("pyarrow", min_version="7.0.0")
+def test_orc_roundtrip_bytesio():
+    # GH44554
+    # PyArrow gained ORC write support with the current argument order
+    data = {
+        "boolean1": np.array([False, True], dtype="bool"),
+        "byte1": np.array([1, 100], dtype="int8"),
+        "short1": np.array([1024, 2048], dtype="int16"),
+        "int1": np.array([65536, 65536], dtype="int32"),
+        "long1": np.array([9223372036854775807, 9223372036854775807], dtype="int64"),
+        "float1": np.array([1.0, 2.0], dtype="float32"),
+        "double1": np.array([-15.0, -5.0], dtype="float64"),
+        "bytes1": np.array([b"\x00\x01\x02\x03\x04", b""], dtype="object"),
+        "string1": np.array(["hi", "bye"], dtype="object"),
+    }
+    expected = pd.DataFrame.from_dict(data)
+
+    bytes = expected.to_orc()
+    got = read_orc(BytesIO(bytes))
+
+    tm.assert_equal(expected, got)
+
+
+@td.skip_if_no("pyarrow", min_version="7.0.0")
+@pytest.mark.parametrize("df_not_supported", orc_writer_dtypes_not_supported)
+def test_orc_writer_dtypes_not_supported(df_not_supported):
+    # GH44554
+    # PyArrow gained ORC write support with the current argument order
+    msg = """The dtype of one or more columns is unsigned integers,
```

Comment:
Nit: Single quotes please

Reply:
Sure!

Reply:
Fixed!

Reply:
Looks like this wasn't changed still.

Reply:
Oops. Now it has been changed.

Reply:
@mroeschke Ah that’s because black replaces single quotes with double ones automatically.